src/core/ready.js

```diff
@@ -49,11 +47,42 @@ jQuery.extend( {
 			return;
 		}
 
-		// If there are functions bound, to execute
-		readyList.resolveWith( document, [ jQuery ] );
+		whenReady = function( fn ) {
+			readyCallbacks.push( fn );
+
+			if ( !readyFiring ) {
+				readyFiring = true;
+
+				// Prevent errors from freezing future callback execution (gh-1823)
+				try {
+					while ( readyCallbacks.length ) {
+						fn = readyCallbacks.shift();
+						if ( jQuery.isFunction( fn ) ) {
+
+							// For backwards compatibility,
+							// invoke synchronously and with document context
+							fn.call( document, jQuery );
+						}
+					}
+				} finally {
+					readyFiring = false;
+
+					// If there was an error in a ready callback,
+					// continue with the rest (gh-1823)
+					if ( readyCallbacks.length ) {
+						whenReady();
```

Comment:
Retrying synchronously here swallows all but the last exception generated by a block of `readyCallbacks` (e.g., `jQuery.ready.then( thrower ); jQuery.ready.then( thrower )`): https://jsfiddle.net/ewjschzw/1/

I think a `setTimeout` is necessary.